File: src/main/java/my/study/repository/design/pattern/decorator/java/ConcreteComponent.java
package my.study.repository.design.pattern.decorator.java;

/**
 * @author djh on  2018/10/2 20:33
 * @E-Mail 1544579459@qq.com
 */
class ConcreteComponent extends BaseComponent {
    @Override
    void operator() {
        System.out.println("基础行为!");
    }
}

File: src/main/java/my/study/repository/design/pattern/decorator/kotlin/ConcreteComponent.kt
package my.study.repository.design.pattern.decorator.kotlin

/**
 * @author djh on  2018/10/2 20:40
 * @E-Mail 1544579459@qq.com
 */
class ConcreteComponent : BaseComponent {
    override fun operator() {
        println("基础行为!")
    }
}